File: runner/LocalRunner.java
import java.io.IOException;
import java.net.ConnectException;
import java.nio.charset.StandardCharsets;
import java.nio.file.Files;
import java.nio.file.Paths;
import java.util.Arrays;
import java.util.HashSet;

public class LocalRunner {
    private static final String LOG_FILE = "out/log.txt";

    private static void run(boolean vis, boolean sync, int teamSize, boolean smartGuy, boolean keyboardPlayer) throws IOException {
        ProcessBuilder b = new ProcessBuilder("java", "-jar", "lib/local-runner.jar",
                String.valueOf(vis),
                String.valueOf(sync),
                String.valueOf(teamSize),
                LOG_FILE,
                String.valueOf(smartGuy),
                String.valueOf(keyboardPlayer)
        );
        b.start();
    }

    /**
     * @param args -vis, -sync, -smartGuy, -keyboard
     */
    public static void main(String[] args) throws IOException, InterruptedException {
        HashSet<String> set = new HashSet<>(Arrays.asList(args));
        run(set.contains("-vis"), set.contains("-sync"), 3, set.contains("-smart"), set.contains("-keyboard"));

        while (true) {
            try {
                Runner.main(new String[]{});
            } catch (ConnectException e) {
                if ("Connection refused".equals(e.getMessage())) {
                    Thread.sleep(500);
                    continue;
                }
            }
            break;
        }

        for (String line : Files.readAllLines(Paths.get(LOG_FILE), StandardCharsets.UTF_8)) {
            System.out.println(line);
        }
    }
}

File: runner/localRunner.kt
package runner.local

import org.apache.log4j.Logger
import java.net.ConnectException
import java.io.File
import runner.time

open class Player(val name: String, val classFile: String)
object MyStrategy : Player("MyStrategy", "#LocalTestPlayer")
object EmptyPlayer : Player("EmptyPlayer", javaClass<com.a.b.a.a.e.b>().getSimpleName() + ".class")
object QuickStartGuy : Player("QuickStartGuy", javaClass<com.a.b.a.a.e.c>().getSimpleName() + ".class")
object SmartGuy : Player("SmartGuy", javaClass<com.a.b.a.a.e.a>().getSimpleName() + ".class")

val LOG_FILE = "out/log.txt"

fun localRunner(vis: Boolean, teamSize: Int, p1: Player, p2: Player, p3: Player, p4: Player): Runnable {
    Logger.getRootLogger()?.removeAllAppenders()

    return com.a.b.c(array(
            "-move-count=50",
            "-render-to-screen=$vis",
            "-render-to-screen-scale=1.0",
            "-render-to-screen-sync=true",
            "-results-file=$LOG_FILE",
            "-debug=true",
            "-base-adapter-port=31000",
            "-seed=",
            "-p1-name=${p1.name}",
            "-p2-name=${p2.name}",
            "-p3-name=${p3.name}",
            "-p4-name=${p4.name}",
            "-p1-team-size=$teamSize",
            "-p2-team-size=$teamSize",
            "-p3-team-size=$teamSize",
            "-p4-team-size=$teamSize",
            p1.classFile,
            p2.classFile,
            p3.classFile,
            p4.classFile
    ))
}

fun runMyStrategy() {
    while (true) {
        try {
            Runner.main(array<String>())
        } catch (e: ConnectException) {
            if (e.getMessage() == "Connection refused") {
                Thread.sleep(200)
                continue
            } else throw e
        }
        break
    }

    println(File(LOG_FILE).readText())
}

fun main(args: Array<String>) = time {
    val p1 = QuickStartGuy
    val p2 = MyStrategy
    val p3 = QuickStartGuy
    val p4 = QuickStartGuy

    Thread(localRunner("-vis" in args.toSet(), 3, p1, p2, p3, p4)).start()

    runMyStrategy()
}
